Mage.Sets/src/mage/cards/a/ApproachOfTheSecondSun.java

```diff
@@ -64,28 +65,28 @@ public ApproachOfTheSecondSunEffect copy() {
     @Override
     public boolean apply(Game game, Ability source) {
         Player controller = game.getPlayer(source.getControllerId());
-        Spell spell = game.getStack().getSpell(source.getSourceId(), false);
-        if (controller != null && spell != null) {
-            ApproachOfTheSecondSunWatcher watcher
-                    = game.getState().getWatcher(ApproachOfTheSecondSunWatcher.class);
-            if (watcher != null
-                    && watcher.getApproachesCast(controller.getId()) > 1
-                    && spell.getFromZone() == Zone.HAND) {
-                // Win the game
-                controller.won(game);
-            } else {
-                // Gain 7 life and put this back into library.
-                controller.gainLife(7, game, source);
-
-                // Put this into the library as the 7th from the top
-                Card spellCard = game.getStack().getSpell(source.getSourceId(), false).getCard();
-                if (spellCard != null) {
+        if (controller == null)
+            return false;
```

Comment:
Shouldn't it also return false if the spell or the watcher are null?

Reply:
@Alex-Vasile the spell is null if the stack item is a copy. Because the flag is set.

Maybe instead, we should just get the spell and check manually if it is a copy? That is how it was before someone refactored the code and created the bug.

Im not sure what would cause the watcher to be null

Reply:
Watcher can't be null in your use case -- it's uses game scope, so will be global for all cards. Null checking for the watcher is a good code style.

Reply:
@Alex-Vasile What might cause a Spell to be null?

Reply:
> @Alex-Vasile What might cause a Spell to be null?

I don't know. I am still not clear on when these can be false. This kind of information we need to gather up and add to a central wiki or somewhere so that others can learn it.

My point was that the original code returned false if the spell is null, but now the code gains them life and goes into the else branch on line 78.

Reply:
> Watcher can't be null in your use case -- it's uses game scope, so will be global for all cards. Null checking for the watcher is a good code style.

I do not understand. But similarly, shouldn't the code return false if the watcher was null, rather than gain life and put the card back in the library?